$ PYTHONPATH=. pytest
================================ test session starts ================================
platform win32 -- Python 3.8.15, pytest-7.1.2, pluggy-1.0.0        
rootdir: C:\Users\Daljeet.Gahle\github\mast-upgrade-baysar-analysis
collected 16 items

tests\test_main.py ..                                                          [ 12%]
tests\OpenADAS\test_get_adf11.py ....                                          [ 37%]
tests\OpenADAS\test_get_adf15.py ......                                        [ 75%]
tests\OpenADAS\test_read_adf15.py ..FF                                         [100%]

===================================== FAILURES ======================================
_____________________________ TestLoadAdf15.test_Ne_III _____________________________

self = <test_read_adf15.TestLoadAdf15 object at 0x00000188FEBAE790>

    def test_Ne_III(self):
        adf15: str = get_adf15(element='ne', charge=2)
>       adf15_model: DataArray = load_adf15(adf15, passed=True)

tests\OpenADAS\test_read_adf15.py:23:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
backend\OpenADAS\read_adf15.py:127: in load_adf15
    adf15_model: DataArray = build_adf15_dataarray(adf15_raw)
backend\OpenADAS\read_adf15.py:76: in build_adf15_dataarray
    total_blocks: int = get_number_of_blocks(adf15_raw)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

adf15_raw = '<!DOCTYPE html \n     PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"\n  
   "http://www.w3.org/TR/xhtml1/DTD/xhtml1-t...as occured. Please try again or contact
 us if the problem\npersists.\n\n</body>\n</html>\n File not found in database.'      

    def get_number_of_blocks(adf15_raw: str) -> int:
        header: str = adf15_raw.split('\n')[0]
>       total_blocks: int = int(
            header.split()[0]
        )
E       ValueError: invalid literal for int() with base 10: '<!DOCTYPE'

backend\OpenADAS\read_adf15.py:20: ValueError
______________________________ TestLoadAdf15.test_H_I _______________________________ 

self = <test_read_adf15.TestLoadAdf15 object at 0x00000188FEBAE8B0>

    def test_H_I(self):
        adf15: str = get_adf15(element='h', charge=0, year=12)
>       adf15_model: DataArray = load_adf15(adf15, passed=True)

tests\OpenADAS\test_read_adf15.py:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
backend\OpenADAS\read_adf15.py:127: in load_adf15
    adf15_model: DataArray = build_adf15_dataarray(adf15_raw)
backend\OpenADAS\read_adf15.py:76: in build_adf15_dataarray
    total_blocks: int = get_number_of_blocks(adf15_raw)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

adf15_raw = '<!DOCTYPE html \n     PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"\n  
   "http://www.w3.org/TR/xhtml1/DTD/xhtml1-t...as occured. Please try again or contact
 us if the problem\npersists.\n\n</body>\n</html>\n File not found in database.'      

    def get_number_of_blocks(adf15_raw: str) -> int:
        header: str = adf15_raw.split('\n')[0]
>       total_blocks: int = int(
            header.split()[0]
        )
E       ValueError: invalid literal for int() with base 10: '<!DOCTYPE'

backend\OpenADAS\read_adf15.py:20: ValueError
============================== short test summary info ============================== 
FAILED tests/OpenADAS/test_read_adf15.py::TestLoadAdf15::test_Ne_III - ValueError: ...
FAILED tests/OpenADAS/test_read_adf15.py::TestLoadAdf15::test_H_I - ValueError: inv...
=========================== 2 failed, 14 passed in 4.74s ============================

--- a/backend/OpenADAS/read_adf15.py
+++ b/backend/OpenADAS/read_adf15.py
@@ -131,7 +131,8 @@ def load_adf15(adf15: [str, Path], passed: bool = False) -> DataArray:
 def read_adf15(adf15: [str, Path],
                block: [int, list],
                ne: ndarray,
-               te: ndarray) -> DataArray:
+               te: ndarray,
+               passed: bool = False) -> DataArray:
     """
     Reads ADAS formatted ADF15 (PEC files) and returns
 
@@ -142,7 +143,7 @@ def read_adf15(adf15: [str, Path],
     :return:
     """
     # Load DataArray
-    adf15_model: DataArray = load_adf15(adf15)
+    adf15_model: DataArray = load_adf15(adf15, passed=passed)
     # Interpolate to get desired values
     block: list = [block] if type(block) is int else block
     kwargs: dict = dict(fill_value="extrapolate")

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,5 @@
 numpy==1.24.1
 pytest==7.1.2
+scipy==1.10.0
 tqdm==4.64.0
 xarray==2022.12.0

--- a/tests/OpenADAS/test_read_adf15.py
+++ b/tests/OpenADAS/test_read_adf15.py
@@ -1,6 +1,8 @@
 # Imports
-from backend.OpenADAS import get_adf15, load_adf15
+from backend.OpenADAS import get_adf15, load_adf15, read_adf15
+from numpy import array, ndarray
 from xarray import DataArray
+
 
 # Variables
 
@@ -26,6 +28,38 @@ class TestLoadAdf15:
     def test_H_I(self):
         adf15: str = get_adf15(element='h', charge=0, year=12)
         adf15_model: DataArray = load_adf15(adf15, passed=True)
+        assert type(adf15_model) is DataArray
+
+
+class TestReadAdf15:
+
+    def test_C_III(self):
+        ne: ndarray = array([1e13, 1e14, 1e15])
+        te: ndarray = array([3, 5, 10], dtype=float)
+        adf15: str = get_adf15(element='c', charge=2)
+        adf15_model: DataArray = read_adf15(adf15, block=1, ne=ne, te=te, passed=True)
+        assert type(adf15_model) is DataArray
+
+    def test_N_II(self):
+        ne: ndarray = array([1e13, 1e14, 1e15])
+        te: ndarray = array([3, 5, 10], dtype=float)
+        adf15: str = get_adf15(element='n', charge=1)
+        adf15_model: DataArray = read_adf15(adf15, block=1, ne=ne, te=te, passed=True)
+        assert type(adf15_model) is DataArray
+
+
+    def test_Ne_III(self):
+        ne: ndarray = array([1e13, 1e14, 1e15])
+        te: ndarray = array([3, 5, 10], dtype=float)
+        adf15: str = get_adf15(element='ne', charge=2)
+        adf15_model: DataArray = read_adf15(adf15, block=1, ne=ne, te=te, passed=True)
+        assert type(adf15_model) is DataArray
+
+    def test_H_I(self):
+        ne: ndarray = array([1e13, 1e14, 1e15])
+        te: ndarray = array([3, 5, 10], dtype=float)
+        adf15: str = get_adf15(element='h', charge=0, year=12)
+        adf15_model: DataArray = read_adf15(adf15, block=1, ne=ne, te=te, passed=True)
         assert type(adf15_model) is DataArray
 
 

